h2o-persist-s3/src/test/java/water/persist/PersistS3Test.java

```diff
@@ -470,5 +475,68 @@ public void testCustomCredentialsProvider_ignoreInvalid() {
     assertEquals(defaultProviders.length, extendedProviders.length);
   }
 
+  @Test
+  public void testImportParquetFromS3() {
+    checkEnv();
+
+    final ArrayList<String> keys = new ArrayList<>();
+    final ArrayList<String> fails = new ArrayList<>();
+    final ArrayList<String> deletions = new ArrayList<>();
+    final ArrayList<String> files = new ArrayList<>();
+
+    try {
+      Scope.enter();
+      H2O.getPM().importFiles("s3://h2o-public-test-data/smalldata/parser/parquet/airlines-simple.snappy.parquet", 
+              null, files, keys, fails, deletions);
 
+      assertEquals(0, fails.size());
+      assertEquals(0, deletions.size());
+      assertEquals(1, files.size()); // Parse including files in sub folders
+      assertEquals(1, keys.size());
+
+      Frame fromS3 = DKV.getGet(keys.get(0));
+      Scope.track(fromS3);
+
+      FileVec localVec = makeNfsFileVec("smalldata/parser/parquet/airlines-simple.snappy.parquet");
+      assertNotNull(localVec);
+      Scope.track(localVec);
+
+      assertEquals(localVec.length(), fromS3.anyVec().length());
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test
+  public void testParseParquetFromS3() throws IOException {
```

Comment:
locally this test is failing for me, looks like it still guesses CSV:
```junit.framework.ComparisonFailure: 
Expected :PARQUET
Actual   :CSV```

Reply:
I think you will need to refresh Idea dependencies in your intellij - most likely it doesn't see Parquet parser (this was added as a test dependency in this PR)

Also, it expects the credentials in environment properties - similarly to other tests.

Reply:
that was it 👌